Cyb xsoar enhancement

suraj-metron opened the pull request:
<!-- REMINDER: THIS IS A PUBLIC REPO DO NOT POST HERE SECRETS/SENSITIVE DATA -->

## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
Added new commands and playbooks

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [x] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [ ] Tests
- [ ] Documentation

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/demisto/content?pullRequest=19493) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/demisto/content?pullRequest=19493) before we can accept your contribution.<br/>**1** out of **3** committers have signed the CLA.<br/><br/>:white_check_mark: suraj-metron<br/>:x: kapil-metron<br/>:x: prasenjit-metron<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/demisto/content?pullRequest=19493) it.</sub>

edik24 commented on Packs/Cybereason/pack_metadata.json:
```diff
@@ -2,7 +2,7 @@
     "name": "Cybereason",
     "description": "Endpoint detection and response to manage and query malops, connections and processes.",
     "support": "xsoar",
```
please change to support: "partner"
and as the pack should be adopted, please follow the adoption guidelines
https://xsoar.pan.dev/docs/partners/adopt#process

edik24 commented on Packs/Cybereason/pack_metadata.json:
```diff
@@ -2,7 +2,7 @@
     "support": "xsoar",
-    "currentVersion": "1.0.20",
+    "currentVersion": "2.0.0",
     "author": "Cortex XSOAR",
```
update to Cybereason

edik24 commented on Packs/Cybereason/pack_metadata.json:
```diff
@@ -2,7 +2,7 @@
-    "currentVersion": "1.0.20",
+    "currentVersion": "2.0.0",
     "author": "Cortex XSOAR",
     "url": "https://www.paloaltonetworks.com/cortex",
```
update cybereason website

edik24 commented on Packs/Cybereason/pack_metadata.json:
```diff
@@ -2,7 +2,7 @@
+    "currentVersion": "2.0.0",
     "author": "Cortex XSOAR",
     "url": "https://www.paloaltonetworks.com/cortex",
     "email": "",
```
add support email

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/Cybereason.py:
```diff
@@ -1381,6 +1630,316 @@ def logout():
+        is_status = malware_status.split(",")
+        query.append({"fieldName": "status", "operator": "Equals", "values": is_status})
+    if bool(time_stamp) is True:
+        query.append({"fieldName": "timestamp", "operator": "GreaterThan", "values": [int(time_stamp)]})
```
Are you sure you want to handle the timestamp without any validation? it will cause unclear errors for bad inputs.

suraj-metron replied:
If there are any unwanted values to it, the API will handle it. Also, the timestamp will be in epoch so we cannot really add any validations here.

JasBeilin replied:
you can use arg_to_number function on it, so it will handle all non-digits/weird chars inputs.

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/Cybereason.py:
```diff
@@ -1381,6 +1630,316 @@ def logout():
+    if bool(malware_status) is True:
+        is_status = malware_status.split(",")
+        query.append({"fieldName": "status", "operator": "Equals", "values": is_status})
+    if bool(time_stamp) is True:
```
```suggestion
    if time_stamp:
```

By replacing it, every non-empty value will count as False.

prasenjit-metron replied:
Resolved.

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/Cybereason.py:
```diff
@@ -1381,6 +1630,316 @@ def logout():
+    if bool(malware_type) is True:
+        types = malware_type.split(",")
+        query.append({"fieldName": "type", "operator": "Equals", "values": types})
+    if bool(malware_status) is True:
```
```suggestion
    if malware_status:
```

prasenjit-metron replied:
Resolved.

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/Cybereason.py:
```diff
@@ -1381,6 +1630,316 @@ def logout():
+
+def malware_query_filter(needs_attention: str, malware_type: str, malware_status: str, time_stamp: str, limit_range: int) -> dict:
+    query = []
+    if bool(needs_attention) is True:
```
```suggestion
    if needs_attention:
```

prasenjit-metron replied:
Resolved.

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/Cybereason.py:
```diff
@@ -1381,6 +1630,316 @@ def logout():
+    query = []
+    if bool(needs_attention) is True:
+        query.append({"fieldName": "needsAttention", "operator": "Is", "values": [bool(needs_attention)]})
+    if bool(malware_type) is True:
```
```suggestion
    if malware_type:
```

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/Cybereason.yml:
```diff
@@ -702,12 +775,298 @@ script:
+      description: The hostname of the machine.
+      isArray: false
+      name: machineName
+      required: false
```
Providing explanation of what happens when no id is specified here would be appriciated by customers.

suraj-metron replied:
We are quite unsure about which id you are referring here, as the command just accepts the machine name as input

JasBeilin replied:
I was referring to the machine name is an identifier, but I saw you handled the case of empty input in the command code, so its ok by me.

JasBeilin reviewed (changes requested):
Thank you for this contribution! 
Please take a look at my comments. In addition, I strongly recommend using the `CommandResult` object instead of `demisto.results()`. I will be happy to help with this change, as I saw this is used in older versions of the pack.   

Also, I saw many new commands, but not tests. Are you planning on adding unittesting for the new command? (I strongly advise you do, as it seems the commands do some data manipulation.

idovandijk reviewed (changes requested):
Thanks for the contribution!
I reviewed the PR and have some points that need your attention:
1. Please change the names to fit the naming standards: `Cybereason - Download Close File` and `Cybereason - Download File`. Also "download close file" is a bit confusing, any better idea?
2. The task inputs are referring to `incident` fields, like `incident.agentid`. To make the playbook more **modular** and **generic**, I highly recommend to do the following:
A. Create a `malopGUID` playbook input.
B. In the `cybereason-start-fetchfile` task, under the `malopGUID` parameter, use `inputs.malopGUID`.
C. Repeat for the other tasks number 1 and 2 in both playbooks
4. In the playbook where you abort the download operation, why do you use `cybereason-start-fetchfile` and then abort it? Shouldn't it be aborting the download for a file that has already been started the fetch process?
5. Does it take time for the file to be downloadable (does it start a scheduled download / job)? If so, don't we need polling to ensure the file can be downloaded? Is polling being done at the code level? 
6. How are the files being made available for download? Are they put in a File war-room entry?

Could you please attach GIFs demonstrating playbook execution and task inputs and results?

JasBeilin commented:
Hi @suraj-metron , any updates? Do you need any help with the changes we asked?

suraj-metron commented:
> Hi @suraj-metron , any updates? Do you need any help with the changes we asked?

Hi @JasBeilin,
Thank you so much for the concern. We are doing changes suggested by you and @idovandijk .
We are replacing all demisto.results() to CommandResults and this process is taking a bit longer. But we will surely fix it soon. If needed, I will surely ask you for help.

ShahafBenYakir commented:
@JasBeilin @suraj-metron what is the status here?

ShahafBenYakir commented:
@suraj-metron are you managing? Is there any need for help from our side?

mchasepan commented:
Just pinged Metron too

suraj-metron commented:
> @suraj-metron are you managing? Is there any need for help from our side?

Hi @ShahafBenYakir ,
We are still in process of making the changes and we are targetting mid next week to complete it.
Thank you for the concern, we really appreciate it!

JasBeilin commented:
Hi @prasenjit-metron , Is the PR finished? Can we start go though it?

prasenjit-metron commented:
> Hi @prasenjit-metron , Is the PR finished? Can we start go though it?

Hi @JasBeilin  , The changes are almost done, we will be done by tomorrow. We will update you once done. Thank you!

prasenjit-metron commented on Packs/Cybereason/Integrations/Cybereason/Cybereason.py:
```diff
@@ -1381,6 +1630,316 @@ def logout():
+        api_response = start_fetchfile(file_guid, user_name)
+        try:
+            if api_response['status'] == "SUCCESS":
+                demisto.results("Successfully started fetching file for the given malop")
```
Resolved.

prasenjit-metron commented on Packs/Cybereason/Integrations/Cybereason/Cybereason.py:
```diff
@@ -1381,6 +1630,316 @@ def logout():
+        'ReadableContentsFormat': formats['text'],
+        'HumanReadable': 'Filename: ' + str(filename) + ' Status: ' + str(status) + ' Batch ID: ' + str(message),
+        'EntryContext': ec
+    })
```
Resolved.

prasenjit-metron commented on Packs/Cybereason/Integrations/Cybereason/Cybereason.py:
```diff
@@ -368,37 +370,43 @@ def query_connections_command():
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': tableToMarkdown('Cybereason Connections for: {}'.format(filter_input), outputs),
+            'EntryContext': ec
+        })
```
Resolved.

prasenjit-metron commented on Packs/Cybereason/Integrations/Cybereason/Cybereason.py:
```diff
@@ -1381,6 +1630,316 @@ def logout():
+        demisto.results(f"Available Sensor IDs are {output}")
+
+
 def main():
```
Resolved.

prasenjit-metron commented on Packs/Cybereason/Integrations/Cybereason/Cybereason.py:
```diff
@@ -111,7 +106,9 @@ def build_query(query_fields, path, template_context='SPECIFIC'):
+        custom_response: bool = False) -> Any:
     LOG('running request with url=%s' % (SERVER + url_suffix))
     try:
         res = session.request(
```
Resolved.

prasenjit-metron commented on Packs/Cybereason/Integrations/Cybereason/Cybereason.yml:
```diff
@@ -702,12 +775,298 @@ script:
+    - contextPath: Download.progress.status
+      description: Status for batch ID
+    - contextPath: Download.progress.batchID
+      description: Unique batch id
```
Resolved.

prasenjit-metron commented on Packs/Cybereason/Integrations/Cybereason/Cybereason.py:
```diff
@@ -1381,6 +1630,316 @@ def logout():
+
+
+def malware_query_command():
+    needs_attention = demisto.getArg('needsAttention')
```
Resolved.

suraj-metron commented on Packs/Cybereason/Integrations/Cybereason/Cybereason.py:
```diff
@@ -1381,6 +1630,316 @@ def logout():
+            if api_response['status'] == "SUCCESS":
+                demisto.results("Successfully started fetching file for the given malop")
+        except Exception:
+            raise Exception("Failed to start fetch file process")
```
yes, we have added this just to handle unwanted situation. Ideally, there will be just one file.

suraj-metron commented on Packs/Cybereason/Integrations/Cybereason/Cybereason.py:
```diff
@@ -1381,6 +1630,316 @@ def logout():
+
+def download_fetchfile_command():
+    batch_id = demisto.getArg('batchID')
+    demisto.log('Downloading the file with this Batch ID: {}'.format(batch_id))
```
We have removed this line, as this was not needed actually.

suraj-metron commented on Packs/Cybereason/Integrations/Cybereason/Cybereason.py:
```diff
@@ -1381,6 +1630,316 @@ def logout():
+
+def malware_query_command():
+    needs_attention = demisto.getArg('needsAttention')
+    malware_type = demisto.getArg('type')
```
The 'type' is not a Boolean variable, it's a string. So we cannot use the argToBoolean function here.

suraj-metron commented:
> Thanks for the contribution! I reviewed the PR and have some points that need your attention:
> 
> 1. Please change the names to fit the naming standards: `Cybereason - Download Close File` and `Cybereason - Download File`. Also "download close file" is a bit confusing, any better idea?
> 2. The task inputs are referring to `incident` fields, like `incident.agentid`.
>    ![image](https://user-images.githubusercontent.com/43602124/176212511-97791095-32e1-41b5-9ba7-472f58cda512.png)
> 
> To make the playbook more **modular** and **generic**, I highly recommend to do the following: A. Create a `malopGUID` playbook input. B. In the `cybereason-start-fetchfile` task, under the `malopGUID` parameter, use `inputs.malopGUID`. C. Repeat for the other tasks number 1 and 2 in both playbooks 3. In the playbook where you abort the download operation, why do you use `cybereason-start-fetchfile` and then abort it? Shouldn't it be aborting the download for a file that has already been started the fetch process? 4. Does it take time for the file to be downloadable (does it start a scheduled download / job)? If so, don't we need polling to ensure the file can be downloaded? Is polling being done at the code level? 5. How are the files being made available for download? Are they put in a File war-room entry?
> 
> Could you please attach GIFs demonstrating playbook execution and task inputs and results?
Answers:
1. Done
2. Done
3. This playbook just demonstrates the use of aborting the process for downloading a file. The user can directly run a single command in War Room to abort/delete a file from server.
4. Yes, it takes time to download the file and we have enabled polling for 60 seconds. For 60 seconds (or any other time specified by the user), the command will poll periodically to check if the progress is completed.


5. Yes, the files are available to download in War Room.

Please find short videos of playbook execution:
Download File: https://user-images.githubusercontent.com/87964764/181027628-04a19cfc-c6f2-4c77-926e-e542a6987954.mp4
Download Close File: https://user-images.githubusercontent.com/87964764/181027559-dc839486-d247-4941-beed-1358d614fb1a.mp4

suraj-metron commented:
Hi @JasBeilin @ShahafBenYakir @idovandijk ,
The PR is updated with the latest changes. Can you please review the changes?
Apologies for the delay caused.

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/Cybereason.py:
```diff
@@ -1106,70 +1269,67 @@ def get_file_machine_details(file_guid):
-        'filterType': 'ContainsIgnoreCase'
-    }]
+def query_domain_command(client: Client, args: dict) -> Any:
+    domain_input_value = args.get('domain')
+    domain_list = domain_input_value.split(",")
```
```suggestion
    domain_list = argToList(args.get('domain'))
```

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/Cybereason.py:
```diff
@@ -1180,64 +1340,63 @@ def query_domain(filters):
+                            username), cybereason_outputs, headers=USER_HEADERS),
+                    outputs_prefix='Cybereason.User',
+                    outputs_key_field='Username',
+                    outputs=ec)
```
```suggestion
                    outputs=cybereason_outputs)
```

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/Cybereason.py:
```diff
@@ -1249,14 +1408,102 @@ def query_user(filters):
+        'POST', '/rest/sensors/action/delete', json_body=data, return_json=False, custom_response=True)
+
+    if response.status_code == 204:
+        output = "The selected Sensor with Sensor ID: {sensor_id} is not available for deleting.".format(sensor_id=sensor_id)
```
```suggestion
        output = f"The selected Sensor with Sensor ID: {sensor_id} is not available for deleting."
```

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/Cybereason.py:
```diff
@@ -1249,14 +1408,102 @@ def query_user(filters):
+    else:
+        try:
+            json_response = response.json()
+            output = "Could not delete Sensor. The received response is {json_response}".format(json_response=json_response)
```
```suggestion
            output = f"Could not delete Sensor. The received response is {json_response}"
```

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/Cybereason.py:
```diff
@@ -1381,85 +1628,459 @@ def logout():
+
+
+def download_fetchfile(client: Client, batch_id: str) -> Any:
+    url = '/rest/fetchfile/getfiles/{batch_id}'.format(batch_id=batch_id)
```
```suggestion
    url = f'/rest/fetchfile/getfiles/{batch_id}'
```

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/Cybereason.py:
```diff
@@ -1381,85 +1628,459 @@ def logout():
+
+
+def close_fetchfile(client: Client, batch_id: str) -> Any:
+    url = '/rest/fetchfile/close/{batch_id}'.format(batch_id=batch_id)
```
```suggestion
    url = f'/rest/fetchfile/close/{batch_id}'
```

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/Cybereason.py:
```diff
@@ -1381,85 +1628,459 @@ def logout():
+    malware_type = args.get('type')
+    malware_status = args.get('status')
+    time_stamp = args.get('timestamp')
+    limit_range = args.get('limit')
```
```suggestion
    limit_range = arg_to_number(args.get('limit'))
```

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/Cybereason.py:
```diff
@@ -1381,85 +1628,459 @@ def logout():
+
+
+def start_host_scan_command(client: Client, args: dict):
+    sensor_id = args.get('sensorID')
+    sensor_ids = sensor_id.split(",")
```
```suggestion
    sensor_ids = argToList(args.get('sensorID'))
```

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/Cybereason.py:
```diff
@@ -1381,85 +1628,459 @@ def logout():
+        'POST', '/rest/sensors/action/schedulerScan', json_body=json_body, return_json=False, custom_response=True)
+    if response.status_code == 204:
+        return CommandResults(
+            readable_output="Given Sensor ID/ID's {sensor_ids} is/are not available for scanning.".format(sensor_ids=sensor_ids))
```
```suggestion
            readable_output=f"Given Sensor ID/ID's {sensor_ids} is/are not available for scanning."
```

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/Cybereason.py:
```diff
@@ -1381,85 +1628,459 @@ def logout():
+        }
+    response = client.cybereason_api_call('POST', '/rest/sensors/query', json_body=json_body)
+    if dict_safe_get(response, ['sensors']) == []:
+        return CommandResults(readable_output="Could not found any Sensor ID for the machine '{}'".format(machine_name))
```
```suggestion
        return CommandResults(readable_output=f"Could not found any Sensor ID for the machine {machine_name}")
```

JasBeilin commented on Packs/Cybereason/Scripts/CybereasonPreProcessingExample/CybereasonPreProcessingExample.py:
```diff
@@ -0,0 +1,21 @@
+from CommonServerPython import *  # noqa: F401
+
+res = True
+incident = demisto.incidents()[0]
```
@idovandijk , when running a pre-process script, do we have the current incident in our hans, meaning demisto.incident() should work?

idovandijk replied:
We would need to check in existing content, not sure

JasBeilin commented on Packs/Cybereason/Scripts/CybereasonPreProcessingExample/CybereasonPreProcessingExample.yml:
```diff
@@ -0,0 +1,22 @@
+enabled: true
+scripttarget: 0
+runonce: false
+runas: DBotWeakRole
```
@idovandijk , When using the DbotWeakRole we had issues with pre-process rules. Do you know if that was solved?

idovandijk replied:
I think Aviad had fixed the issue but also not sure

JasBeilin reviewed (changes requested):
You did great job here! 
I have a few comments:
1. I see some function do not have unitest but contain fragile logic, such as update_output. also non of your new command has tests, and since there is a lot of logic in how you show the result and the calculation of the results, it requires adding some good tests here. When running demisto-sdk lint you can see a coverage report, we require 75% percent of tests.

2. I see you moved to python3 which is awesome. When using python3 you can use formatted strings (I changed some for you, not all.) using f-strings is much more clear. Please change the .format to f-string.

3. When using the CommandResult you do not need to specify the context path and id, as it is built-in in the commandResult (the `Cybereason.Machine(val.Name && val.Name === obj.Name)` part). I changed it for you. Please make sure it is still showing the results as you wished.

idovandijk reviewed (commented):
As for the codeless parts I'm approving this, @JasBeilin when code review is done feel free to approve.

Well done @suraj-metron!

suraj-metron commented:
> You did great job here! I have a few comments:
> 
> 1. I see some function do not have unitest but contain fragile logic, such as update_output. also non of your new command has tests, and since there is a lot of logic in how you show the result and the calculation of the results, it requires adding some good tests here. When running demisto-sdk lint you can see a coverage report, we require 75% percent of tests.
> 2. I see you moved to python3 which is awesome. When using python3 you can use formatted strings (I changed some for you, not all.) using f-strings is much more clear. Please change the .format to f-string.
> 3. When using the CommandResult you do not need to specify the context path and id, as it is built-in in the commandResult (the `Cybereason.Machine(val.Name && val.Name === obj.Name)` part). I changed it for you. Please make sure it is still showing the results as you wished.

Hi @JasBeilin,
Thank you for your kind words!
We will make the changes as suggested by you and will update you soon on this one.
Thank you so much for suggesting the changes.

suraj-metron commented:
> As for the codeless parts I'm approving this, @JasBeilin when code review is done feel free to approve.
> 
> Well done @suraj-metron!

Thank you @idovandijk for your kind words.
We really appreciate your feedback.

suraj-metron commented:
> You did great job here! I have a few comments:
> 
> 1. I see some function do not have unitest but contain fragile logic, such as update_output. also non of your new command has tests, and since there is a lot of logic in how you show the result and the calculation of the results, it requires adding some good tests here. When running demisto-sdk lint you can see a coverage report, we require 75% percent of tests.
> 2. I see you moved to python3 which is awesome. When using python3 you can use formatted strings (I changed some for you, not all.) using f-strings is much more clear. Please change the .format to f-string.
> 3. When using the CommandResult you do not need to specify the context path and id, as it is built-in in the commandResult (the `Cybereason.Machine(val.Name && val.Name === obj.Name)` part). I changed it for you. Please make sure it is still showing the results as you wished.

Hi @JasBeilin,
We had a quick update on the changes:
We have resolved the 2nd and 3rd comment. As of now we are working on the unit tests and we will get it resolved as soon as possible.

Thanks

suraj-metron commented:
Hi @JasBeilin ,
Had a quick update, we have added unit tests which takes the coverage to 60%. As of now, we are covering different edge cases to increase it to 75%.

suraj-metron commented:
Hi @JasBeilin 
We have updated the PR and fixed all the given comments. Please review it.
Thank you!

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/test_files/fetch_scan_status_raw_response.json:
```diff
@@ -0,0 +1,8415 @@
+    },
+    "finalState": "true",
+    "totalNumberOfProbes": 1,
+    "initiatorUser": "prashant@metronlabs.com",
```
Please use mocked email such example@example.com

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/test_files/fetch_scan_status_raw_response.json:
```diff
@@ -0,0 +1,8415 @@
+    "abortTime": 0,
+    "abortTimeout": "false",
+    "abortHttpStatusCode": null,
+    "creatorUser": "prashant@metronlabs.com"
```
Please use mocked email such example@example.com

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/test_files/fetch_scan_status_raw_response.json:
```diff
@@ -0,0 +1,8415 @@
+    },
+    "finalState": "true",
+    "totalNumberOfProbes": 1,
+    "initiatorUser": "gal.turgeman@safebreach.com",
```
Please use mocked email such example@example.com

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/test_files/fetch_scan_status_raw_response.json:
```diff
@@ -0,0 +1,8415 @@
+    "abortTime": 0,
+    "abortTimeout": "false",
+    "abortHttpStatusCode": null,
+    "creatorUser": "gal.turgeman@safebreach.com"
```
Please use mocked email such example@example.com

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/test_files/fetch_scan_status_raw_response.json:
```diff
@@ -0,0 +1,8415 @@
+    },
+    "finalState": "true",
+    "totalNumberOfProbes": 1,
+    "initiatorUser": "prashant@metronlabs.com",
```
Please use mocked email such example@example.com

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/test_files/fetch_scan_status_raw_response.json:
```diff
@@ -0,0 +1,8415 @@
+    "abortTime": 0,
+    "abortTimeout": "false",
+    "abortHttpStatusCode": null,
+    "creatorUser": "prashant@metronlabs.com"
```
Please use mocked email such example@example.com

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/test_files/fetch_scan_status_raw_response.json:
```diff
@@ -0,0 +1,8415 @@
+    },
+    "finalState": "true",
+    "totalNumberOfProbes": 1,
+    "initiatorUser": "prashant@metronlabs.com",
```
Please use mocked email such example@example.com

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/test_files/fetch_scan_status_raw_response.json:
```diff
@@ -0,0 +1,8415 @@
+    "abortTime": 0,
+    "abortTimeout": "false",
+    "abortHttpStatusCode": null,
+    "creatorUser": "prashant@metronlabs.com"
```
Please use mocked email such example@example.com

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/test_files/fetch_scan_status_raw_response.json:
```diff
@@ -0,0 +1,8415 @@
+    },
+    "finalState": "true",
+    "totalNumberOfProbes": 1,
+    "initiatorUser": "prashant@metronlabs.com",
```
Please use mocked email such example@example.com

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/test_files/fetch_scan_status_raw_response.json:
```diff
@@ -0,0 +1,8415 @@
+    "abortTime": 0,
+    "abortTimeout": "false",
+    "abortHttpStatusCode": null,
+    "creatorUser": "prashant@metronlabs.com"
```
Please use mocked email such example@example.com

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/test_files/fetch_scan_status_raw_response.json:
```diff
@@ -0,0 +1,8415 @@
+    },
+    "finalState": "true",
+    "totalNumberOfProbes": 1,
+    "initiatorUser": "prashant@metronlabs.com",
```
Please use mocked email such example@example.com

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/test_files/fetch_scan_status_raw_response.json:
```diff
@@ -0,0 +1,8415 @@
+    "abortTime": 0,
+    "abortTimeout": "false",
+    "abortHttpStatusCode": null,
+    "creatorUser": "prashant@metronlabs.com"
```
Please use mocked email such example@example.com

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/test_files/fetch_scan_status_raw_response.json:
```diff
@@ -0,0 +1,8415 @@
+    "abortTime": 0,
+    "abortTimeout": "false",
+    "abortHttpStatusCode": null,
+    "creatorUser": "prashant@metronlabs.com"
```
Please use mocked email such example@example.com

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/test_files/fetch_scan_status_raw_response.json:
```diff
@@ -0,0 +1,8415 @@
+    },
+    "finalState": "true",
+    "totalNumberOfProbes": 1,
+    "initiatorUser": "prashant@metronlabs.com",
```
Please use mocked email such example@example.com

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/test_files/fetch_scan_status_raw_response.json:
```diff
@@ -0,0 +1,8415 @@
+    "abortTime": 0,
+    "abortTimeout": "false",
+    "abortHttpStatusCode": null,
+    "creatorUser": "prashant@metronlabs.com"
```
Please use mocked email such example@example.com

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/test_files/fetch_scan_status_raw_response.json:
```diff
@@ -0,0 +1,8415 @@
+    },
+    "finalState": "true",
+    "totalNumberOfProbes": 1,
+    "initiatorUser": "prashant@metronlabs.com",
```
Please use mocked email such example@example.com

JasBeilin reviewed (changes requested):
That is some amazing job!

I did have some comments, most of which changing playbook's id (and name) or changing to mock ip/emails in the test data.
also, please change the name of the `test_files` folder to 'test_data'. These actions which will allow the validation on the PR pass.

I also see the build is failing due to your unittest failing. If you need any assistance in fixing it let me know.

Overall great job, I hope we can move forward in the upcoming days :)

suraj-metron commented:
> That is some amazing job!
> 
> I did have some comments, most of which changing playbook's id (and name) or changing to mock ip/emails in the test data. also, please change the name of the `test_files` folder to 'test_data'. These actions which will allow the validation on the PR pass.
> 
> I also see the build is failing due to your unittest failing. If you need any assistance in fixing it let me know.
> 
> Overall great job, I hope we can move forward in the upcoming days :)

Hi @JasBeilin 
Thank you for your feedback. We have done all the required changes. Please review the PR again!

JasBeilin commented:
Looks great!
I would like to schedule a short demo just to see the commands running as expected so we can merge.

suraj-metron commented:
> Looks great! I would like to schedule a short demo just to see the commands running as expected so we can merge.

Thanks @JasBeilin , we will be ready for the demo at the scheduled meeting.

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/Cybereason.py:
```diff
@@ -1381,85 +1574,456 @@ def logout():
+        file_download = fileResult('download.zip', response.content)
+        return file_download
+    else:
+        return CommandResults(
```
```suggestion
        raise DemistoException("request failed with the following error: " + response.content + " Response Status code: " + str(response.status_code))```

JasBeilin commented:
Post Demo we asked for a few last changes: 
1. Change error messages to actual error (which will be handled in the main function)
2. CommandResult should have outputs and not ec for context view.
3. Change outputs to markdown (using MarkdownToTable function) if possible to show "prettier" results.

JasBeilin commented:
Hi @suraj-metron,
How are you doing? Do you need any assistance?

suraj-metron commented:
> Hi @suraj-metron, How are you doing? Do you need any assistance?

Hi @JasBeilin 
I am doing good with the changes. Thank you for asking.
Just testing all the existing features to make sure nothing got broke with the changes

suraj-metron commented:
Hi @JasBeilin 
Thank you for your patience.
We have done the suggested changes.
Please review.

JasBeilin reviewed (approved):
Looks good to me! You did amazing job here.

suraj-metron commented:
> @suraj-metron Looks good to me! You did amazing job here. Please merge from master on your side so the build will pass and we can move forward :)

Thank you @JasBeilin for your kind words.
I have merged the master brach. The conflicts are resolved now.

JasBeilin commented:
Hi @suraj-metron, It looks like some of the recent changes failed the build (The CybereasonPreProcessingExample lost it's main function for some reason, for example)

Let me know if you need any assistance in fixing it.

I also saw you are using sleep, though we agreed in the start it is not to be used as it is not supported in the XSOAR. Let me know if there is any other possible solution. If not, we may consider using polling command instead.

suraj-metron commented:
> Hi @suraj-metron, It looks like some of the recent changes failed the build (The CybereasonPreProcessingExample lost it's main function for some reason, for example)
> 
> Let me know if you need any assistance in fixing it.
> 
> I also saw you are using sleep, though we agreed in the start it is not to be used as it is not supported in the XSOAR. Let me know if there is any other possible solution. If not, we may consider using polling command instead.

Hi @JasBeilin 
Thank you for pointing it out. We will soon fix this issues and update you.

bakatzir commented:
Hey team, our mutual customers are eager to see this pack updates through.
Let me know what I can do to expedite this.
@johestephan FYI

suraj-metron commented:
> Hey team, our mutual customers are eager to see this pack updates through. Let me know what I can do to expedite this. @johestephan FYI

Hi @bakatzir
We are working on the changes on priority and will soon update this.
Thank you !

suraj-metron commented:
Hi @JasBeilin 
For the sleep statements, we need to have it because we are trying to download a file from server.
To check the status of the file download we need the sleep statements as we need to check the progress of download at particular intervals.
The sleep statements are for 10 seconds.
We have added a default timeout of 60 seconds for this sleep statements. So, after 60 seconds if the file is still not downloaded the command will be exited automatically. It wont run in loop for forever.
Thank you!

JasBeilin commented:
> Hi @JasBeilin For the sleep statements, we need to have it because we are trying to download a file from server. To check the status of the file download we need the sleep statements as we need to check the progress of download at particular intervals. The sleep statements are for 10 seconds. We have added a default timeout of 60 seconds for this sleep statements. So, after 60 seconds if the file is still not downloaded the command will be exited automatically. It wont run in loop for forever. Thank you!

We have discussed it before and this can be solved using polling command. 
We do not support sleep statement and it may cause server issues to clients.

suraj-metron commented:
> > Hi @JasBeilin For the sleep statements, we need to have it because we are trying to download a file from server. To check the status of the file download we need the sleep statements as we need to check the progress of download at particular intervals. The sleep statements are for 10 seconds. We have added a default timeout of 60 seconds for this sleep statements. So, after 60 seconds if the file is still not downloaded the command will be exited automatically. It wont run in loop for forever. Thank you!
> 
> We have discussed it before and this can be solved using polling command. We do not support sleep statement and it may cause server issues to clients.

Hi @JasBeilin ,
Sure we will use the Polling commands instead of sleep statements. Meanwhile, we have fixed all other major errors and pushed the code.
Once the polling changes are done, we will update you again.

Thank you!

suraj-metron commented:
Hi @JasBeilin 
We have fixed the major errors because of which the build was failing earlier.
Can you please confirm if we need to make some other changes as well?

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/Cybereason.py:
```diff
@@ -225,7 +228,7 @@ def get_machine_guid(client: Client, machine_name: str) -> str:
 
 def is_probe_connected_command(client: Client, args: dict, is_remediation_commmand: bool = False) -> Any:
-    machine = args.get('machine')
+    machine = str(args.get('machine'))
```
No need to convert to a string as the demisto.args() returning all the arguments as str already. Did you encounter any issue with it?

suraj-metron replied:
Yes, we were getting a MyPy error if we didn't added the typecast. Please let me know if you want us to remove it.

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/Cybereason.py:
```diff
@@ -270,7 +273,7 @@ def is_probe_connected(client: Client, machine: str) -> dict:
 
 def query_processes_command(client: Client, args: dict):
-    machine = args.get('machine')
+    machine = str(args.get('machine'))
```
No need to convert to a string as the demisto.args() returning all the arguments as str already. Did you encounter any issue with it?

suraj-metron replied:
Yes, we were getting a MyPy error if we didn't added the typecast. Please let me know if you want us to remove it.

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/Cybereason.py:
```diff
@@ -308,7 +311,7 @@ def query_processes_command(client: Client, args: dict):
     return CommandResults(
         readable_output=tableToMarkdown('Cybereason Processes', outputs, headers=PROCESS_HEADERS),
-        outputs_prefix='Cybereason.Process',
+        outputs_prefix='Process',
```
I strongly advice against this change. Think about a case where a client is using both you product and another one that can search for processes. Then the client needs to be able to know what came from your integration and what came for another one.

suraj-metron replied:
Sure Jawine.
That makes complete sense. We will add the Cybereason Prefix again.
Thanks for pointing it out.

suraj-metron replied:
Fixed now

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/Cybereason.py:
```diff
@@ -401,7 +404,7 @@ def query_connections_command(client: Client, args: dict):
         return CommandResults(
             readable_output=tableToMarkdown(f'Cybereason Connections for: {filter_input}', outputs),
-            outputs_prefix='Cybereason.Connection',
+            outputs_prefix='Connection',
```
I strongly advice against this change. Think about a case where a client is using both you product and another one that can search for connection. Then the client needs to be able to know what came from your integration and what came for another one.

suraj-metron replied:
Sure Jawine.
That makes complete sense. We will add the Cybereason Prefix again.
Thanks for pointing it out.

suraj-metron replied:
Fixed now

JasBeilin commented:
@suraj-metron overall it looks good.
I saw that you added casting to string in some places, why is that? It should be automatic, did you have any issues with it?
Also, I see you changed the context path of some of the outputs. I really think you need to think this change through.

suraj-metron commented:
Hi @JasBeilin,
I have made the changes.
Can you please review the PR again?
Thanks!

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/Cybereason.py:
```diff
@@ -723,363 +716,466 @@ def malop_processes(malop_guids):
+    user_name = str(args.get('userName'))
+    comment = str(args.get('comment')) if args.get('comment') else 'Kill Process Remediation Action Succeeded'
+    remediation_action = 'KILL_PROCESS'
+    is_machine_conntected = is_probe_connected_command(client, args, is_remediation_commmand=True)
+    if is_machine_conntected is True:
```
```suggestion
    is_machine_connected = is_probe_connected_command(client, args, is_remediation_command=True)
    if is_machine_connected is True:
```

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/Cybereason.py:
```diff
@@ -723,363 +716,466 @@ def malop_processes(malop_guids):
+    user_name = str(args.get('userName'))
+    comment = str(args.get('comment')) if args.get('comment') else 'Quarantine File Remediation Action Succeeded'
+    remediation_action = 'QUARANTINE_FILE'
+    is_machine_conntected = is_probe_connected_command(client, args, is_remediation_commmand=True)
+    if is_machine_conntected is False:
```
```suggestion
    is_machine_connected = is_probe_connected_command(client, args, is_remediation_command=True)
    if is_machine_connected is False:
```

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/Cybereason.py:
```diff
@@ -723,363 +716,466 @@ def malop_processes(malop_guids):
+    user_name = str(args.get('userName'))
+    comment = str(args.get('comment')) if args.get('comment') else 'Unquarantine File Remediation Action Succeded'
+    remediation_action = 'UNQUARANTINE_FILE'
+    is_machine_conntected = is_probe_connected_command(client, args, is_remediation_commmand=True)
+    if is_machine_conntected is True:
```
```suggestion
    is_machine_connected = is_probe_connected_command(client, args, is_remediation_command=True)
    if is_machine_connected is True:
```

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/Cybereason.py:
```diff
@@ -723,363 +716,466 @@ def malop_processes(malop_guids):
+    user_name = str(args.get('userName'))
+    comment = str(args.get('comment')) if args.get('comment') else 'Block File Remediation Action Succeeded'
+    remediation_action = 'BLOCK_FILE'
+    is_machine_conntected = is_probe_connected_command(client, args, is_remediation_commmand=True)
+    if is_machine_conntected is True:
```
```suggestion
    is_machine_connected = is_probe_connected_command(client, args, is_remediation_command=True)
    if is_machine_connected is True:
```

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/Cybereason.py:
```diff
@@ -723,363 +716,466 @@ def malop_processes(malop_guids):
+    user_name = str(args.get('userName'))
+    comment = str(args.get('comment')) if args.get('comment') else 'Delete Registry Key Remediation Action Succeeded'
+    remediation_action = 'DELETE_REGISTRY_KEY'
+    is_machine_conntected = is_probe_connected_command(client, args, is_remediation_commmand=True)
+    if is_machine_conntected is True:
```
```suggestion
    is_machine_connected = is_probe_connected_command(client, args, is_remediation_command=True)
    if is_machine_connected is True:
```

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/Cybereason.py:
```diff
@@ -723,363 +716,466 @@ def malop_processes(malop_guids):
+    user_name = str(args.get('userName'))
+    comment = str(args.get('comment')) if args.get('comment') else 'Kill Prevent Unsuspend Remediation Action Succeeded'
+    remediation_action = 'KILL_PREVENT_UNSUSPEND'
+    is_machine_conntected = is_probe_connected_command(client, args, is_remediation_commmand=True)
+    if is_machine_conntected is True:
```
```suggestion
    is_machine_connected = is_probe_connected_command(client, args, is_remediation_command=True)
    if is_machine_connected is True:
```

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/Cybereason.py:
```diff
@@ -723,363 +716,466 @@ def malop_processes(malop_guids):
+    user_name = str(args.get('userName'))
+    comment = str(args.get('comment')) if args.get('comment') else 'Unsuspend Process Remediation Action Succeeded'
+    remediation_action = 'UNSUSPEND_PROCESS'
+    is_machine_conntected = is_probe_connected_command(client, args, is_remediation_commmand=True)
+    if is_machine_conntected is True:
```
```suggestion
    is_machine_connected = is_probe_connected_command(client, args, is_remediation_command=True)
    if is_machine_connected is True:
```

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/Cybereason.py:
```diff
@@ -188,51 +218,44 @@ def get_machine_guid(machine_name):
 
-def is_probe_connected_command(is_remediation_commmand=False):
-    machine = demisto.getArg('machine')
+def is_probe_connected_command(client: Client, args: dict, is_remediation_commmand: bool = False) -> Any:
```
```suggestion
def is_probe_connected_command(client: Client, args: dict, is_remediation_command: bool = False) -> Any:
```

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/Cybereason.py:
```diff
@@ -188,51 +218,44 @@ def get_machine_guid(machine_name):
+            is_connected = True
+            break
 
     if is_remediation_commmand:
```
```suggestion
    if is_remediation_command:
```

JasBeilin reviewed (changes requested):
I found some small misspelled names.
Also, you have some flake8 erros. Please fix thouse.

suraj-metron commented:
> I found some small misspelled names. Also, you have some flake8 erros. Please fix thouse.

Thank you so much for pointing out the misspelled names. We will fix those along with the Flake8 errors

JasBeilin commented:
Hi @suraj-metron, are you finished with changes? can I do one last review?

suraj-metron commented:
> Hi @suraj-metron, are you finished with changes? can I do one last review?

Hi @JasBeilin 
Yes, we are done with the changes. We were stuck with something but now it is resolved.
Can you please review the PR again?
Thank you!

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/Cybereason_test.py:
```diff
@@ -137,9 +138,10 @@ def test_malop_processes_command(mocker):
     mocker.patch.object(demisto, 'results')
-    with pytest.raises(Exception):   
+    with pytest.raises(Exception):
         command_output = malop_processes_command(client, args)
```
I suggest checking the content of the exception to make sure the function failed for the right reasons.

suraj-metron replied:
Done

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/Cybereason_test.py:
```diff
@@ -502,32 +459,12 @@ def test_quarantine_file_command(mocker):  # _with_connection
     with pytest.raises(Exception):
-        command_output = quarantine_file_command(client, args)
+        quarantine_file_command(client, args)
 
```
I suggest checking the content of the exception to make sure the function failed for the right reasons.

suraj-metron replied:
Done

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/Cybereason_test.py:
```diff
@@ -548,30 +485,10 @@ def test_unquarantine_file_command(mocker):  # _with_connection
     raw_response = json.loads(load_mock_response('get_remediation_action_status.json'))
     mocker.patch("Cybereason.get_remediation_action_status", return_value=raw_response)
     with pytest.raises(Exception):
```
I suggest checking the content of the exception to make sure the function failed for the right reasons.

suraj-metron replied:
Done

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/Cybereason_test.py:
```diff
@@ -590,31 +507,12 @@ def test_block_file_command(mocker):  # _with_connection
     raw_response = json.loads(load_mock_response('get_remediation_action_status.json'))
-    mocker.patch("Cybereason.get_remediation_action_status", return_value=raw_response)    
+    mocker.patch("Cybereason.get_remediation_action_status", return_value=raw_response)
     with pytest.raises(Exception):
```
I suggest checking the content of the exception to make sure the function failed for the right reasons.

suraj-metron replied:
Done

JasBeilin commented on Packs/Cybereason/Scripts/CybereasonPreProcessingExample/CybereasonPreProcessingExample.py:
```diff
@@ -2,26 +2,27 @@
+    incident['id'] = malop_incident['id']
+    demisto.executeCommand('setIncident', incident)
 
-if __name__ in ['__main__', '__builtin__', 'builtins']:
-    main()
```
Why did we remove the main structure?

suraj-metron replied:
It was not required, since whatever we are calling now everything is decalred outside of main. So we removed it.

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/Cybereason.py:
```diff
@@ -9,17 +9,14 @@
 import time
 import re
 
```
```suggestion
import urllib3
```

Due to an upgrade of MyPy this will fix the build fails

suraj-metron replied:
Done

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/Cybereason.py:
```diff
@@ -9,17 +9,14 @@
-    del os.environ['https_proxy']
-
 # disable insecure warnings
 requests.packages.urllib3.disable_warnings()
```
```suggestion
urllib3.disable_warnings()
```
Due to an upgrade of MyPy this will fix the build fails

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/Cybereason.py:
```diff
@@ -553,94 +559,89 @@ def query_malops(total_result_limit=None, per_group_limit=None, template_context
+def malop_processes_command(client: Client, args: dict):
+    malop_guids = args.get('malopGuids')
+    machine_name = str(args.get('machineName'))
+    date_time = str(args.get('dateTime'))
+
+    filter_input = []
+    if date_time:
```
This is a logical issue-
if you don't get dateTime, the get('dateTime') will return None, which will be converted to 'None' due to the str() method.

So in that case, the `if` statement will actually ask `if "None"` and since this is not an empty string, it will return True when you expect False.
I would recommend adding a unitest for this use-case.

suraj-metron replied:
> @suraj-metron, please let me know when you finished with my suggestion/ comments.

Sure @JasBeilin . I will definitely update you once we are done

suraj-metron replied:
Thank you for poiting this out. We have fixed this issue as well as tested it against the Unit Test

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/Cybereason.py:
```diff
@@ -553,94 +559,89 @@ def query_malops(total_result_limit=None, per_group_limit=None, template_context
+
+    filter_input = []
+    if date_time:
+        date_time = dateparser.parse(date_time).timestamp()
```
I would suggest using a new name for this variable, as date_time is str type by your first definition and it changes its type on the go.

suraj-metron replied:
Done

JasBeilin commented on Packs/Cybereason/Scripts/CybereasonPreProcessingExample/CybereasonPreProcessingExample.yml:
```diff
@@ -17,6 +17,6 @@ scripttarget: 0
+- Cybereason Pre Processing Test
 fromversion: 5.0.0
-dockerimage: demisto/python3:3.10.6.33415
+dockerimage: demisto/python3:3.10.7.35188
```
```suggestion
dockerimage: demisto/python3:3.10.8.36650
```

JasBeilin commented:
@suraj-metron, please let me know when you finished with my suggestion/ comments.

suraj-metron commented on Packs/Cybereason/Integrations/Cybereason/Cybereason.py:
```diff
@@ -1765,25 +1767,27 @@ def malware_query_command(client: Client, args: dict):
-    if limit_range > 0:
-        filter_response = malware_query_filter(client, needs_attention, malware_type, malware_status, time_stamp, limit_range)
-        return CommandResults(raw_response=filter_response)
+    if limit_range:
+        if limit_range > 0:
```
Hi @JasBeilin ,
The extra If is added to avoid negative values.
If the limit_range is negative, the condition if limit_range will be true and will continue further. So that's why we have added an extra If.

suraj-metron commented on Packs/Cybereason/Integrations/Cybereason/Cybereason_test.py:
```diff
@@ -109,12 +110,12 @@ def test_two_query_file(mocker):
     mocker.patch("Cybereason.Client.cybereason_api_call", return_value={})
     with pytest.raises(Exception):
-        command_output = query_file_command(client, args)
+        query_file_command(client, args)
```
Done

suraj-metron commented on Packs/Cybereason/Integrations/Cybereason/Cybereason_test.py:
```diff
@@ -109,12 +110,12 @@ def test_two_query_file(mocker):
     mocker.patch("Cybereason.Client.cybereason_api_call", return_value=raw_response)
     with pytest.raises(Exception):
-        command_output = query_file_command(client, args)
+        query_file_command(client, args)
```
Done

suraj-metron commented:
Hi @JasBeilin 
We have completed the changes suggested by you.
Please review the PR again.

Thanks!

edik24 commented:
Hi @suraj-metron I understand that there was a discussion between metron (Prashant) and Cybereason, and it was agreed that cybereason will adopt the pack, so can you please edit the metadata file:
support - must say partner
author - must say your company name
url - must be changed to your company’s support site
email - must be your company's support email.

thanks,

suraj-metron commented:
> Hi @suraj-metron I understand that there was a discussion between metron (Prashant) and Cybereason, and it was agreed that cybereason will adopt the pack, so can you please edit the metadata file: support - must say partner author - must say your company name url - must be changed to your company’s support site email - must be your company's support email.
> 
> thanks,

Hi @edik24,
Yes, you are correct about the adopt the pack process.
Thank you for mentioning it.
I have updated the necessary files for adopt a pacl process.
Please review the changes and let me know if any more changes are needed.

Thanks!

JasBeilin commented on Packs/Cybereason/Integrations/Cybereason/Cybereason.py:
```diff
@@ -553,94 +560,89 @@ def query_malops(total_result_limit=None, per_group_limit=None, template_context
+
+    filter_input = []
+    if date_time != 'None':
+        date_time_parser = dateparser.parse(date_time).timestamp()
```
```suggestion
        date_time_parser = dateparser.parse(date_time)
        if not date_time_parser:
            raise DemistoException("dateTime could not be parsed. Please enter a valid time parameter.")
        date_time_parser = date_time_parser.timestamp()
```

JasBeilin reviewed (changes requested):
Great.
Please take a look at my last fix suggestion, make sure it is ok on your side. Once commited the build should pass and we can merge :)

suraj-metron commented:
> Great. Please take a look at my last fix suggestion, make sure it is ok on your side. Once commited the build should pass and we can merge :)

Thanks @JasBeilin for suggesting the change.
We have done the changes and tested the related command again.
It is working fine.
Please review the PR again

JasBeilin commented:
Great, we are good to go!
Since you take ownership, please note that we need a partner image:
`Partners must provide a non-empty author image under the path /home/circleci/project/Packs/Cybereason/Author_image.png`.
Please provide it and we will merge :)
@edik24 FYI

suraj-metron commented:
> @suraj-metron Great, we are good to go! Since you take ownership, please note that we need a partner image: `Partners must provide a non-empty author image under the path /home/circleci/project/Packs/Cybereason/Author_image.png`. Please provide it and we will merge :) @edik24 FYI

Hi @JasBeilin,
Sure, we will soon add the Author Image and once done we will update you.
Thank you!

suraj-metron commented:
Hi @JasBeilin 
We have added Author Image in our Pack.
Please review the PR again.
Thanks!
